### PR TITLE
PWX-26175: Adding license-extension tests

### PR DESCRIPTION
**Pull request by zoxpx**
Adding set of Torpedo for the license-extension

Additional:
* fixed version-check for master-branch
* adding `RunCommandInPodEx()` command

Signed-off-by: Zoran Rajic <zrajic@purestorage.com>

**What this PR does / why we need it**:

This adds the "blue-green license extension" integration tests --
* since we have to control the PX-cluster formation, the operator integration-tests were more appropriate than Torpedo-tests

**Which issue(s) this PR fixes** (optional)
Closes # PWX-26175

**Special notes for your reviewer**:

Note that we introduced the new `RunCommandInPodEx()` command --
* note that this change in this PR is only temporary
* we will add this change into `sched-ops` package, and "vendor it back in"  (see https://github.com/portworx/sched-ops/pull/302)

### CODE REVIEW

File: test/integration_test/bluegreen_test.go
Comment (pure-jliao) [inline test/integration_test/bluegreen_test.go:109]: nit: we can remove those should skip comments.

**Review by pure-jliao — COMMENTED** [review]
each test cases in the list should be designed to run individually given a storage cluster spec and validate different functionalities, but seems they are more like steps now, which is kind of strange.. we can call `tc.TestSpec(t)` in each test cases, if the stc name is not changed the cluster should be reused. but if this test has its dedicated pipeline job and meets your requirement, I feel it's ok.

File: vendor/github.com/portworx/sched-ops/k8s/core/pods.go
Comment (zoxpx) [reply vendor/github.com/portworx/sched-ops/k8s/core/pods.go:73]: PR already posted at https://github.com/portworx/sched-ops/pull/302  (still waiting for approval)

**Comment by zoxpx** [comment]
> each test cases in the list should be designed to run individually given a storage cluster spec and validate different functionalities, but seems they are more like steps now, which is kind of strange.. we can call `tc.TestSpec(t)` in each test cases, if the stc name is not changed the cluster should be reused. but if this test has its dedicated pipeline job and meets your requirement, I feel it's ok.

Well.. I really needed a _sequence_ of inter-dependent tests, e.g.
1. install cluster + license for 3 nodes
2. attempt adding 4th node into the cluster  (expect failure, clean up 4th node)
3. activate license-extension
4. attempt adding 4th node  (should work this time)

I could have packaged the whole sequence in a single huge test  (and imo, make a big mess out of it) -- or add it as a separate tests 1..4, and expect they'd be ran in a sequence.

**Comment by zoxpx** [comment]
Note, with [6619eec](https://github.com/libopenstorage/operator/pull/938/commits/6619eece8d5cc858ad84f3a32030369eb101b277) we have a successful Jenkins-driven test.

Also note that the review-feedback was given for "vendored in" `.RunCommandInPodEx()` -- so I changed it first here, so we can test it with this integration test  (will re-sync to https://github.com/portworx/sched-ops/pull/302)

**Comment by zoxpx** [comment]
Note, with [54e9a7e](https://github.com/libopenstorage/operator/pull/938/commits/54e9a7e7e664f0b601baf9b41675f3cd745ca721) I've brought in the latest `sched-ops` package, and the dependent https://github.com/portworx/sched-ops/pull/302 fix
* note that I also adjusted the go.mod golang version to `go 1.18`

**Review by pure-jliao — APPROVED** [review]
LGTM, sorry for the long waiting

**Comment by zoxpx** [comment]
Woops.. looks like I forgot to commit this PR.

**Pull request merged**
